src/System.Private.CoreLib/shared/System/IO/TextWriter.cs

```diff
@@ -595,8 +596,13 @@ public Task WriteAsync(char[]? buffer)
             return WriteAsync(buffer, 0, buffer.Length);
         }
 
-        public virtual Task WriteAsync(char[] buffer, int index, int count)
+        public virtual Task WriteAsync(char[]? buffer, int index, int count)
         {
+            if (buffer == null)
```

Comment:
This needs to check arguments. If buffer is null, index and count better be null as well.

Reply:
Right, will fix.